Fix tests

thomasritz opened the pull request:
This pull request fixes a couple of failing tests and removes some warnings.

Furthermore, it removes Ruby 1.9.2 from the Travis test suite.
And it completes the test matrix to test both Ruby 2.0.0 and 2.2.2 with ActiveResource 4.0, 4.1 and 4.x.

thomasritz committed:
update `appraisal`, remove some warnings, travis w/o ruby 1.9.2, fix tests

1 test still fails

thomasritz committed:
use webmock to count requests

apepper commented on Gemfile:
```diff
@@ -4,16 +4,11 @@ source 'https://rubygems.org'
-if RUBY_VERSION =~ /^1\.9/
-  gem 'json'
-  gem "redcarpet"
-  gem "yard"
```
Does yard documentation still works? Currently there is http://www.rubydoc.info/gems/infopark_crm_connector/1.2.0 .

thomasritz replied:
I don't know. I removed Ruby 1.9 from the tests. If Ruby 1.9 is still required for some tasks, it should be documented so. I'm going to revert those lines if it's required for yard…

apepper replied:
`yard` and `redcarpet` are probably required. I don't know, why there are only in the ruby 1.9 block. Must be a bug.

apepper commented on Gemfile:
```diff
@@ -4,16 +4,11 @@ source 'https://rubygems.org'
 gem "rake"
 gem "rest-client"
 gem "test-unit"
 gem "webmock"
```
Please limit the version of `webmock`.

thomasritz replied:
I haven't introduced `webmock`. Please add the version you would like.

apepper commented:
FEEDBACK (098a724)

Thank you for your contribution.

Here are some issues with this PR:
* [ ] Please ensure, that `yard` is still working (see details at https://github.com/infopark/crm_connector/pull/19#discussion-diff-78544045 ).
* [ ] Please limit the version of `webmock`, e.g. `~> 2.1` (see details at https://github.com/infopark/crm_connector/pull/19#discussion-diff-78544174).

thomasritz committed:
gems for yard

thomasritz committed:
update versions in readme

thomasritz committed:
test building the yard doc on travis

thomasritz committed:
Limit webmock to version 2.x

thomasritz committed:
yard obviously requires ruby 1.9.3

thomasritz committed:
mime-types-data requires Ruby version >= 2.0

thomasritz committed:
dont' run cleanup for the yard task

thomasritz commented:
@apepper I limited the webmock version to 2.1., I reverted the yard gemfile lines and added a yard task to the travis build (build it with Ruby 2.0.0). Travis is green. Since RubyDoc.info is powered by yard 0.9.5 (ruby-2.3.1) I hope yard doc works again.

apepper commented:
OK (221e898)

Great job 🎉
